File: 1758-minimum-changes-to-make-alternating-binary-string/1758-minimum-changes-to-make-alternating-binary-string.go
func calc(s string, start int) int {
    res := 0
    goal := start

    for _. ch := range s {
        if byte(ch) - '0' != goal {
            res++
        }
        goal ^= 1
    }

    return res
}

func minOperations(s string) int {
    ops := calc(s, 0)
    return min(len(s) - ops) 
}

File: 1758-minimum-changes-to-make-alternating-binary-string/1758-minimum-changes-to-make-alternating-binary-string.cpp
class Solution {
public:
    int calc(string &s, int start) {
        int res = 0;
        int goal = start;
        for(char ch : s) {
            int curr = ch - '0';
            if(curr != goal) res++;
            goal ^= 1;
        }

        return res;
    }
    int minOperations(string s) {
        return min(calc(s, 0), calc(s, 1));
    }
};